src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/SpanNDHelpers.T.cs

```diff
@@ -0,0 +1,311 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+
+
+#pragma warning disable 8500 // sizeof of managed types
+
+namespace System.Numerics.Tensors
+{
+    internal static partial class SpanHelpers // .T
+    {
+        public static unsafe void Memmove<T>(Span<T> destination, ReadOnlySpan<T> source, nint length, nint dstOffset = 0)
```

Comment:
Why is all of this here? Why can't any need for this just rely on MemoryExtensions?

Reply:
So `System.Span` uses `Buffer.Memmove` and `Buffer._Memmove`, both of which are internal so @tannergooding suggested using `NativeMemory.Copy`.

When I have an array for the underlying data I could use `MemoryExtensions.CopyTo`, but when I don't have an array for the underlying data I can't (unless I am just not seeing an overload). Currently `Tensor<T>` does have the underlying array, but we know that in the future we will need to support native memory as well (either with another Tensor class or something), as well as the `SpanND` not having an array for the underlying data. So instead of making 2 different methods to handle both cases I just made 1 that will work for both.

If you would prefer me to use `MemoryExtensions.CopyTo` for the Tensor I can, but Spans will still have to be done this way unless you can point me to the extension/overload I am missing.

Reply:
I'd prefer we not duplicate all this complicated and unsafe code and just use `Span<T>`/`MemoryExtensions`. Those are all based on `int`, but to support larger `nuint` ranges we can just partition it up into `int.MaxValue`-sized chunks and delegate for each, as I did for Clear in the previous comment. Memmove is a tad trickier if we have to handle overlapping regions, but that really just means comparing the references / bounds and having two different loops, one for copying forwards and one for copying backwards.

Reply:
Sorry, I hadn't actually scrolled down. These Memmoves are fine. But everything after that should be deleted, e.g. if you need Fill, that can be done just like I showed for Clear.

Reply:
Right. My suggestion to use `NativeMemory.Copy` was specifically to avoid duplicating large amounts of code and to avoid needing to also break it up into row by row chunks for contiguous spans.

The worst case scenario for a lot of these should be grabbing the longest contiguous region of the slice and doing the operation on that, then repeating across the rest of the higher dimension counts. For something that retains the row-major ordering of the strides (that is `_strides` are functionally sorted) this is incredibly trivial/efficient.

Reply:
This particular API is just doing `Span.Copy` and so I don't think we need to defer to NativeMemory.Copy here.

Reply:
So I have made the changes to Clear/Fill/SequenceEqual.

@tannergooding do you think I should also rewrite the memmove stuff to use SPan.CopyTo?

Reply:
That's probably best overall.

I expect that the majority of tensors we interact with will be views/slices and the individual contiguous parts will then fit into Span, so there's no real benefit to deferring to the APIs that explicitly work with large data.